packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -355,12 +356,13 @@ class _ModalBottomSheetRoute<T> extends PopupRoute<T> {
   final double elevation;
   final ShapeBorder shape;
   final Clip clipBehavior;
+  final bool isDismissible;
 
   @override
   Duration get transitionDuration => _bottomSheetDuration;
 
   @override
-  bool get barrierDismissible => true;
+  bool get barrierDismissible => isDismissible ?? true;
```

Comment:
This should then just be 

```suggestion
  bool get barrierDismissible => isDismissible;
```

Reply:
Done.